src/main/java/net/fabricmc/loader/impl/ModContainerImpl.java

```diff
@@ -30,11 +37,22 @@
 public class ModContainerImpl extends net.fabricmc.loader.ModContainer {
 	private final LoaderModMetadata info;
 	private final Path originPath;
+	private final String parentModId;
+	private final Collection<String> childModIds;
+
 	private volatile Path root;
 
-	public ModContainerImpl(LoaderModMetadata info, Path originPath) {
-		this.info = info;
-		this.originPath = originPath;
+	public ModContainerImpl(ModCandidate candidate) {
+		this.info = candidate.getMetadata();
+		this.originPath = candidate.getPath();
+		this.parentModId = candidate.getParentMods().isEmpty() ? null : candidate.getParentMods().iterator().next().getId();
```

Comment:
Is it well defined that the cadinate's first parent mod will always be the first one in the collection?

Reply:
No, but the logic there disambiguates it. This relates to how ModDiscoverer deals with identical nested jars: If two mods have the same nested jar by size+hash, loader will read it only once and attaches it to both containing mods. There is no meaningful difference between the parent choices.